fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/PushDownPredicateAggRule.java

```diff
@@ -44,26 +51,100 @@ public List<OptExpression> transform(OptExpression input, OptimizerContext conte
                 iter.remove();
                 // and to predicates
                 pushDownPredicates.add(scalar);
+
+                if (Utils.isConstantEqualPredicate(scalar)) {
+                    constantEquivalentPredicate.put((ColumnRefOperator) scalar.getChild(0), scalar.getChild(1));
+                }
             }
         }
 
         // merge filter
         filters.add(logicalAggOperator.getPredicate());
-        input.setChild(0, OptExpression.create(new LogicalAggregationOperator.Builder().withOperator(logicalAggOperator)
-                        .setPredicate(Utils.compoundAnd(filters))
-                        .build(),
-                input.inputAt(0).getInputs()));
-
-        // push down
-        if (pushDownPredicates.size() > 0) {
-            LogicalFilterOperator newFilter = new LogicalFilterOperator(Utils.compoundAnd(pushDownPredicates));
-            OptExpression oe = new OptExpression(newFilter);
-            oe.getInputs().addAll(input.inputAt(0).getInputs());
-
-            input.inputAt(0).getInputs().clear();
-            input.inputAt(0).getInputs().add(oe);
+        ScalarOperator scalarOperator = Utils.compoundAnd(filters);
+        ScalarOperatorRewriter scalarRewriter = new ScalarOperatorRewriter();
+        // The calculation of the null value is in the constant fold
+        if (scalarOperator != null) {
+            scalarOperator = scalarRewriter.rewrite(scalarOperator, ScalarOperatorRewriter.DEFAULT_REWRITE_RULES);
         }
 
-        return Lists.newArrayList(input.inputAt(0));
+        // If there is a constant equivalent predicate, delete the column ref corresponding to
+        // the constant in the aggregation node
+        // (because the column ref must be a constant, the aggregation of constants has no meaning
+        // unless the constant is the only grouping key)
+        if (!constantEquivalentPredicate.isEmpty()) {
```

Comment:
I think would better put this logic to another new class.

Reply:
At present, only in agg push down, it can be judged that there is a constant predicate push down, and it can be judged that a certain column in agg is a constant, and other positions cannot associate agg and filter.